internal/plugin/installer/plugin_structure_test.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright The Helm Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package installer
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestDetectPluginRoot(t *testing.T) {
+	tests := []struct {
+		name        string
+		setup       func(dir string) error
+		expectRoot  string
+		expectError bool
+	}{
+		{
+			name: "plugin.yaml at root",
+			setup: func(dir string) error {
+				return os.WriteFile(filepath.Join(dir, "plugin.yaml"), []byte("name: test"), 0644)
+			},
+			expectRoot:  ".",
+			expectError: false,
+		},
+		{
+			name: "plugin.yaml in subdirectory",
+			setup: func(dir string) error {
+				subdir := filepath.Join(dir, "my-plugin")
+				if err := os.MkdirAll(subdir, 0755); err != nil {
+					return err
+				}
+				return os.WriteFile(filepath.Join(subdir, "plugin.yaml"), []byte("name: test"), 0644)
+			},
+			expectRoot:  "my-plugin",
+			expectError: false,
+		},
```

Comment:
```suggestion
		},
				{
			name: "multiple plugin.yamls in subdirectories",
			setup: func(dir string) error {
				subdir := filepath.Join(dir, "plugins", "plugin1")
				if err := os.MkdirAll(subdir, 0755); err != nil {
					return err
				}
				os.WriteFile(filepath.Join(subdir, "plugin.yaml"), []byte("name: test"), 0644)
				subdir = filepath.Join(dir, "plugins", "plugin2")
				if err := os.MkdirAll(subdir, 0755); err != nil {
					return err
				}
				return os.WriteFile(filepath.Join(subdir, "plugin.yaml"), []byte("name: test2"), 0644)
			},
			expectRoot:  "my-plugin",
			expectError: false,
		},
```

In case you wanted to be more explicit about supporting having multiple folders each with a plugin.yaml, but only getting the first one.

Reply:
I just think we should align plugin archives with how chart archives are handled.